sys/matstat/matstat.c

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright (C) 2018 Eistec AB
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <stdint.h>
+#include "matstat.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+void matstat_clear(matstat_state_t *state)
+{
+    *state = MATSTAT_STATE_INIT;
+}
+
+void matstat_add(matstat_state_t *state, int32_t value)
+{
+    if (value > state->max) {
+        state->max = value;
+    }
+    if (value < state->min) {
+        state->min = value;
+    }
+    state->sum += value;
+    if (state->count == 0) {
+        state->offset = value;
+    }
+    /* Subtracting this offset yields better numeric stability in the variance
+     * algorithm */
+    value -= state->offset;
+    /* Assuming the differences will be small on average and the number of
+     * samples is reasonably limited, to prevent overflow in sum_sq */
+    state->sum_sq += (int64_t)value * value;
+    ++state->count;
```

Comment:
I would prefer `++state->count` just after the `state->sum += value`.
This way, the first part of the function handles the min/max/mean and the second part the variance and so the `offset` and `sum_sq` values.